src/Widget/Field.php

```diff
@@ -737,21 +744,34 @@ private function addInputId(): string
     }
 
     /**
-     * Set form model, name and options for the widget.
+     * Set form model and name for the widget.
      *
      * @param FormModelInterface $data Form model.
      * @param string $attribute Form model property this widget is rendered for.
-     * @param array $options The HTML attributes for the widget container tag.
-     * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      *
      * @return self
      */
-    public function config(FormModelInterface $data, string $attribute, array $options = []): self
+    public function config(FormModelInterface $data, string $attribute): self
     {
         $new = clone $this;
         $new->data = $data;
         $new->attribute = $attribute;
+        return $new;
+    }
+
+    /**
+     * Generate a container tag for {@see attribute}.
+     *
+     * @param bool $containerEnabled enabled/disable container for the widget.
+     * @param array $options The HTML attributes for the widget container tag if it is enabled.
+     * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self
+     */
+    public function enclosedByContainer($containerEnabled = true, array $options = []): self
+    {
+        $new = clone $this;
         $new->options = $options;
```

Comment:
Why did you move `$options` here?

Reply:
See [discussion](https://github.com/yiisoft/form/pull/33#discussion_r469294033) on @terabytesoftw requested changes above.

Reply:
What about to split these onto 2 independent methods?

Reply:
Do you mean something like `containerEnabled(bool $enabled)` and `containerOptions(array $options = [])`?
I have tried to replicate `label(bool $enabledLabel = true, array $options = [], ?string $label = null)` method, which is already there and combines enabling/disabling label with its' options.

Reply:
I'm ok with any approach allowing to disable container, but I would like to keep behaviors persistent. So in case of 2 independent methods, the same thing must be done for label.

Reply:
Ok let it be. We can think about it later.

Reply:
It can be split to withoutContainer() and withContainer(array $options).

Reply:
@samdark What about label in that case?

Reply:
That's just a method split nothing is changed about label.

Reply:
@samdark I mean the same must be done with `label` method (which works the same now) to be consistent.